app/bundles/EmailBundle/EventListener/MatchFilterForLeadTrait.php

```diff
@@ -151,23 +151,34 @@ protected function matchFilterForLead(array $filter, array $lead)
                     break;
                 case 'in':
                     $leadValMatched = false;
-                    foreach ($leadVal as $k => $v) {
-                        if (in_array($v, $filterVal)) {
+                    if (is_array($leadVal)) {
+                        foreach ($leadVal as $k => $v) {
+                            if (in_array($v, $filterVal)) {
+                                $leadValMatched = true;
+                                // Break once we find a match
+                                break;
+                            }
+                        }
+                    } else {
+                        if (in_array($leadVal, $filterVal)) {
```

Comment:
There is some duplication of code. How about to just add this if:
```
if (!is_array($leadVal)) {
    $leadVal = [$leadVal];
}
```
and then continue with the original foreach?